cmd/diff.go

```diff
@@ -0,0 +1,359 @@
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			flags, err := getDiffCommandFlags(cmd, tracker, cmdEvent)
+			if err != nil {
+				return err
+			}
+
+			execx.Debug = flags.Debug
+
+			kubeconfigPath := flags.Kubeconfig
+
+			if kubeconfigPath == "" {
+				kubeconfigFromEnv := os.Getenv("KUBECONFIG")
+
+				if kubeconfigFromEnv == "" {
+					return ErrKubeconfigReq
+				}
+
+				kubeconfigPath = kubeconfigFromEnv
+
+				logrus.Warnf("Missing --kubeconfig flag, falling back to KUBECONFIG from environment: %s", kubeconfigFromEnv)
+			}
+
+			kubeAbsPath, err := filepath.Abs(kubeconfigPath)
+			if err != nil {
+				return fmt.Errorf("error while getting absolute path of kubeconfig: %w", err)
+			}
+
+			kubeconfigPath = kubeAbsPath
+
+			if _, err := os.Stat(kubeconfigPath); os.IsNotExist(err) {
+				return fmt.Errorf("%w in %s", ErrKubeconfigNotFound, kubeconfigPath)
+			}
+
+			logrus.Debug("Getting Home Directory Path...")
+			outDir := flags.Outdir
+
+			homeDir, err := os.UserHomeDir()
+			if err != nil {
+				cmdEvent.AddErrorMessage(err)
+				tracker.Track(cmdEvent)
+
+				return fmt.Errorf("error while getting user home directory: %w", err)
+			}
+
+			if outDir == "" {
+				outDir = homeDir
+			}
+
+			if flags.BinPath == "" {
+				flags.BinPath = filepath.Join(outDir, ".furyctl", "bin")
+			}
+
+			client := netx.NewGoGetterClient()
+			distrodl := distribution.NewDownloader(client, flags.HTTPS)
+
+			logrus.Info("Downloading distribution...")
+			res, err := distrodl.Download(flags.DistroLocation, flags.FuryctlPath)
+			if err != nil {
+				cmdEvent.AddErrorMessage(err)
+				tracker.Track(cmdEvent)
+
+				return fmt.Errorf("error while downloading distribution: %w", err)
+			}
+
+			basePath := filepath.Join(outDir, ".furyctl", res.MinimalConf.Metadata.Name)
+
+			stateStore := state.NewStore(
+				res.RepoPath,
+				flags.FuryctlPath,
+				kubeconfigPath,
+				basePath,
+				res.DistroManifest.Tools.Common.Kubectl.Version,
+				flags.BinPath,
+			)
+
+			diffChecker, err := createDiffChecker(stateStore, flags.FuryctlPath)
+			if err != nil {
+				cmdEvent.AddErrorMessage(err)
+				tracker.Track(cmdEvent)
+
+				return fmt.Errorf("error while creating diff checker: %w", err)
+			}
+
+			phasePath, err := getPhasePath(
+				flags.FuryctlPath,
+				basePath,
+				res.RepoPath,
+				flags.BinPath,
+				kubeconfigPath,
+				flags.Phase,
+				res.MinimalConf,
+				res.DistroManifest,
+			)
+			if err != nil {
+				cmdEvent.AddErrorMessage(err)
+				tracker.Track(cmdEvent)
+
+				return fmt.Errorf("error while getting phase path: %w", err)
+			}
+
+			d, err := getDiffs(diffChecker, phasePath)
+			if err != nil {
+				cmdEvent.AddErrorMessage(err)
+				tracker.Track(cmdEvent)
+
+				return fmt.Errorf("error while getting diffs: %w", err)
+			}
+
+			if len(d) > 0 {
+				logrus.Infof(
+					"Differences found from previous cluster configuration:\n%s",
+					diffChecker.DiffToString(d),
```

Comment:
I would not use the logger(which traditionally writes to stderr) for the actual output of the program. I would use something that targets stdout, such as `fmt.Println()`